frontend/src/pages/articles/my-articles.tsx

```diff
@@ -1,12 +1,54 @@
+import {
+  useAppDispatch,
+  useAppSelector,
+  useCallback,
+  useEffect,
+} from '~/libs/hooks/hooks.js';
+import { type UserDetailsResponseDto } from '~/packages/users/users.js';
+import { actions as articlesActions } from '~/slices/articles/articles.js';
+
 import { ArticleCard } from './components/components.js';
-import { MOCKED_ARTICLES, MOCKED_USER } from './libs/constants.js';
-
-const MyArticles: React.FC = () => (
-  <>
-    {MOCKED_ARTICLES.map((article) => (
-      <ArticleCard key={article.id} article={article} user={MOCKED_USER} />
-    ))}
-  </>
-);
+import { MOCKED_REACTIONS, MOCKED_TAGS } from './libs/constants.js';
+
+const MyArticles: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const { articles, user } = useAppSelector((state) => ({
+    articles: state.articles.articles,
+    user: state.auth.user,
+  }));
+
+  const userDetails = {
+    firstName: user?.firstName,
+    lastName: user?.lastName,
+  } as UserDetailsResponseDto;
+
+  const handleGetMyArticles = useCallback(
+    (id: number) => {
+      void dispatch(articlesActions.loadOwn({ id }));
+    },
+    [dispatch],
+  );
+
+  useEffect(() => {
+    if (user) {
+      handleGetMyArticles(user.id);
+    }
+  }, [handleGetMyArticles, user]);
+
+  return (
+    <>
+      {articles &&
+        articles.map((article) => (
+          <ArticleCard
+            key={article.id}
+            article={article}
+            user={userDetails}
+            tags={MOCKED_TAGS}
+            reactions={MOCKED_REACTIONS}
+          />
+        ))}
```

Comment:
Empty array is truthy value - this check won't do anything
```suggestion
      {articles.map((article) => (
          <ArticleCard
            key={article.id}
            article={article}
            user={userDetails}
            tags={MOCKED_TAGS}
            reactions={MOCKED_REACTIONS}
          />
        ))}
```

Reply:
Should I completely remove the check or just add the length check for the articles?